### PR TITLE
{Network} `az network application-gateway waf-policy policy-setting update`: Add support for log scrubbing and inspection limit

### PR DESCRIPTION
**Pull request by ZengTaoxu**
**Related command**

**Description**

Resolve: #26002
Resolve: #26377

**Testing Guide**

**History Notes**

[Network] `az network application-gateway waf-policy policy-setting update`: Add support for log scrubbing
[Network] `az network application-gateway waf-policy policy-setting update`: Add support for inspection limit
[Network] BREAKING CHANGE: `az network application-gateway waf-policy`: Rename output property `applicationGatewayIpConfigurations` to `applicationGatewayIPConfigurations` to keep consistent with the name in API

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py [hunk 1]
```diff
@@ -103,12 +108,68 @@ def _build_arguments_schema(cls, *args, **kwargs):
             default=False,
             nullable=True,
         )
+        _args_schema.request_body_enforcement = AAZBoolArg(
+            options=["--request-body-enforce", "--request-body-enforcement"],
+            help="Whether allow WAF to enforce request body limits.",
+            nullable=True,
+        )
+        _args_schema.request_body_inspect_limit_in_kb = AAZIntArg(
+            options=["--rbi-limit-in-kb", "--request-body-inspect-limit-in-kb"],
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py:117]: Is `--rbi-limit-in-kb` an official name?

File: src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py [hunk 2]
```diff
@@ -103,12 +108,68 @@ def _build_arguments_schema(cls, *args, **kwargs):
...
+            nullable=True,
+        )
+        _element.selector_match_operator = AAZStrArg(
+            options=["selector-match-operator"],
+            help="When matchVariable is a collection, operate on the selector to specify which elements in the collection this rule applies to.",
+            enum={"Equals": "Equals", "EqualsAny": "EqualsAny"},
+        )
+        _element.state = AAZStrArg(
+            options=["state"],
+            help="Defines the state of log scrubbing rule. Default value is Enabled.",
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py:169]:
```suggestion
            help="Define the state of log scrubbing rule. Default value is Enabled.",
```

File: src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py [hunk 3]
```diff
@@ -2142,7 +2235,7 @@ def _build_schema_subnet_read(cls, _schema):
             serialized_name="addressPrefixes",
         )
         properties.application_gateway_ip_configurations = AAZListType(
-            serialized_name="applicationGatewayIpConfigurations",
+            serialized_name="applicationGatewayIPConfigurations",
```
Comment (necusjz) [inline src/azure-cli/azure/cli/command_modules/network/aaz/latest/network/application_gateway/waf_policy/policy_setting/_update.py:2238]: Breaking change should be added to channgelog.

**Review by necusjz — APPROVED** [review]

**Pull request merged**